src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMIT.java

```diff
@@ -66,6 +69,43 @@ public void testCheckout() throws Exception {
         assertTrue(build.getWorkspace().child("add_file").isDirectory());
     }
 
+    @Test
+    public void testCheckoutAndPush() throws Exception {
+        project.setScm(createSCM("*/master"));
+        String uniqueMessage = UUID.randomUUID().toString();
+        Shell postScript = new Shell("git checkout master\n" +
+                                     "echo \"Hello, World!\" >> test.txt\n" +
+                                     "git add test.txt\n" +
+                                     "git commit -m \"" + uniqueMessage + "\"\n" +
+                                     "git push");
+
+        project.getBuildersList().add(postScript);
+        project.save();
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        BitbucketRevisionAction revisionAction = build.getAction(BitbucketRevisionAction.class);
+        String restPath = new StringBuilder().append(BitbucketUtils.BITBUCKET_BASE_URL)
+                .append("/rest/api/1.0/projects/")
+                .append(BitbucketUtils.PROJECT_KEY)
+                .append("/repos/")
+                .append(BitbucketUtils.REPO_SLUG)
+                .append("/commits?since=")
+                .append(revisionAction.getRevisionSha1())
+                .toString();
+
+        assertEquals(SUCCESS, build.getResult());
```

Comment:
/nit I think this would read better if the path variable definition and the rest call using it was not broken up with this assertion.

Reply:
Agreed. I also inlined the rest path, which I think makes it a bit easier to read.